beacon-chain/rpc/eth/blob/handlers.go

```diff
@@ -80,32 +80,65 @@ func (s *Server) Blobs(w http.ResponseWriter, r *http.Request) {
 				http2.HandleError(w, "blobs are not supported before Deneb fork", http.StatusBadRequest)
 				return
 			}
-			sidecars, err = s.BeaconDB.BlobSidecarsBySlot(r.Context(), primitives.Slot(slot), indices...)
+			ok, roots, err := s.BeaconDB.BlockRootsBySlot(ctx, primitives.Slot(slot))
+			if !ok {
+				http2.HandleError(w, fmt.Sprintf("no block roots at slot %d", slot), http.StatusBadRequest)
+				return
+			}
 			if err != nil {
-				http2.HandleError(w, errors.Wrapf(err, "could not retrieve blobs for slot %d", slot).Error(), http.StatusInternalServerError)
+				http2.HandleError(w, errors.Wrap(err, "failed to get block roots by slot").Error(), http.StatusInternalServerError)
 				return
 			}
-			http2.WriteJson(w, buildSidecarsResponse(sidecars))
+			root = roots[0][:]
+			if len(roots) == 1 {
+				break
+			}
+			for _, blockRoot := range roots {
```

Comment:
taken from the block root endpoint we check if it's canonical as there might be multiple roots